### ISSUE TITLE
Use site date format in invoice.php template for payment date

### ISSUE THREAD

**Issue by nosegraze** [restrictcontentpro/restrict-content-pro#1664]
Right now we're using:

```php
date_i18n( 'dS F, Y', strtotime( $rcp_payment->date, current_time( 'timestamp' ) ) )
```

But there's [a note in the code reference for this function](https://developer.wordpress.org/reference/functions/date_i18n/) that says:

> Note that the “format”, however, is not converted to a local one if you manually specify it. If you need a localized format, then you should use `get_option('date_format')` if you need the format set by the user in Settings->General, and thus one of their choosing. Alternatively, you can wrap your predefined format in `__()` in order to allow translators to adjust the date to the proper local format. If you do so, then you should also include a translator comment, to let the translators know what the date format is referring to and where it is used, so they can convert it accurately.

tl;dr we need to change `'dS F, Y'` to `get_option( 'date_format' )`
